src/components/space/ChartsAlerts.js

```diff
@@ -0,0 +1,73 @@
+import React, { useContext } from 'react';
+import Alert from '@material-ui/lab/Alert';
+import { makeStyles } from '@material-ui/core/styles';
+import { useTranslation } from 'react-i18next';
+import { SpaceDataContext } from '../../contexts/SpaceDataProvider';
+
+const useStyles = makeStyles((theme) => ({
+  root: {
+    flexGrow: 1,
+    paddingLeft: theme.spacing(2),
+    paddingRight: theme.spacing(2),
+    marginBottom: theme.spacing(2),
+  },
+  alert: {
+    display: 'flex',
+    alignItems: 'center',
+    marginBottom: theme.spacing(1),
+  },
+}));
+
+function ChartsAlerts() {
+  const { t } = useTranslation();
+  const classes = useStyles();
+  const { metadata } = useContext(SpaceDataContext);
+  const {
+    maxTreeLength,
+    maxTreeLengthExceeded,
+    requestedSampleSize,
+    numActionsRetrieved,
+  } = metadata;
+
+  const displayWarningMessage = () => {
+    if (numActionsRetrieved === 0) {
+      return (
+        <Alert severity="warning" className={classes.alert}>
+          {t('This space and its subspaces do not have any actions yet.')}
+        </Alert>
+      );
+    }
+    if (maxTreeLengthExceeded) {
+      return (
+        <Alert severity="warning" className={classes.alert}>
+          {t(
+            `This space has more than ${maxTreeLength} subspaces. The data below is sampled from only the first ${maxTreeLength} retrieved subspaces.`,
+          )}
+        </Alert>
+      );
+    }
+    return null;
+  };
+
+  const displaySampleInfo = () => {
+    if (numActionsRetrieved !== 0) {
+      return (
+        <Alert severity="info" className={classes.alert}>
+          {t(
+            `Displaying ${numActionsRetrieved} actions from a requested sample of ${requestedSampleSize} actions.`,
```

Comment:
In order to translate with variables, read the documentation from React i18next.

Reply:
A quick and dirty workaround is to split into multiple translations.

Reply:
Interesting. You're right. Interpolating variables stops the translations from working. I used the quick and dirty workaround.